source/general/entity/damage_source.json

```diff
@@ -3,42 +3,41 @@
   "description": "The types of damage an entity can receive.",
   "type": "string",
   "enum": [
-    "all",
-    "anvil",
-    "block_explosion",
-    "campfire",
-    "charging",
-    "contact",
-    "drowning",
-    "entity_attack",
-    "entity_explosion",
-    "fall",
-    "falling_block",
-    "fatal",
-    "fire",
-    "fire_tick",
-    "fireworks",
-    "fly_into_wall",
-    "freezing",
-    "lava",
-    "lightning",
-    "magic",
-    "magma",
-    "none",
-    "override",
-    "piston",
-    "projectile",
-    "ram_attack",
-    "sonic_boom",
-    "soul_campfire",
-    "stalactite",
-    "stalagmite",
-    "starve",
-    "suffocation",
-    "suicide",
-    "temperature",
-    "thorns",
-    "void",
+    "all",	
+    "anvil",	
+    "block_explosion",	
+    "campfire",	
+    "charging",	
+    "contact",	
+    "drowning",	
+    "entity_attack",	
+    "entity_explosion",	
+    "fall",	
+    "falling_block",	
+    "fire",	
+    "fire_tick",	
+    "fireworks",	
+    "fly_into_wall",	
+    "freezing",	
+    "lava",	
+    "lightning",	
+    "magic",	
+    "magma",	
+    "none",	
+    "override",	
+    "piston",	
+    "projectile",	
+    "ram_attack",	
+    "self_destruct",	
+    "sonic_boom",	
+    "soul_campfire",	
+    "stalactite",	
+    "stalagmite",	
+    "starve",	
+    "suffocation",	
+    "temperature",	
+    "thorns",	
+    "void",	
     "wither"

```

Comment:
Can you revert this change? The list should be as fully defined as possible 🤔 .

We will use the diagnoser or other methods to remove / or notify about is allowed or not. As Minecraft might decide to do support it for other things in the future. Or we can more clearly explain why something cant be used at a certain point

Reply:
Its still mentioned on https://learn.microsoft.com/en-us/minecraft/creator/reference/content/entityreference/examples/filters/has_damage?view=minecraft-bedrock-stable

Reply:
It's in has_damage but it's not listed under the actual damage sources list, I've accounted for that and the enum in has damage include it